dockstore-webservice/src/main/java/io/dockstore/webservice/core/Version.java

```diff
@@ -219,6 +223,13 @@
     @BatchSize(size = 25)
     private Set<Image> images = new HashSet<>();
 
+    @OneToMany(fetch = FetchType.LAZY, orphanRemoval = true, cascade = CascadeType.ALL)
+    @JoinTable(name = "version_metrics", joinColumns = @JoinColumn(name = "versionid", referencedColumnName = "id", columnDefinition = "bigint"), inverseJoinColumns = @JoinColumn(name = "metricsid", referencedColumnName = "id", columnDefinition = "bigint"))
+    @MapKeyColumn(name = "platform")
+    @MapKeyEnumerated(EnumType.STRING)
+    @ApiModelProperty(value = "The aggregated metrics for executions of this version, grouped by platform", position = 26)
+    private Map<Partner, Metrics> metricsByPlatform = new EnumMap<>(Partner.class);
```

Comment:
Just started reviewing, so this may become apparent later in the review, but why is the map key a `Partner`, but the description and field refer to `Platform`?

Also, TIL, there is an `EnumMap` class 👍

Reply:
I'm re-using the `Partner` enum that [`CloudInstance` uses](https://github.com/dockstore/dockstore/blob/0ff1f507610a0ddf125b2f3781c0bda3992a846e/dockstore-webservice/src/main/java/io/dockstore/webservice/core/CloudInstance.java#L47) because they seem to represent the same thing. We've always used `platform` when referring to where the workflow is executed when discussing metrics. I thought about changing the existing `Partner` enum name to `Platform`, but I think that will have ripple effects in the UI so I didn't. Do you think I should change the name in the description/field to `partner` instead?

Reply:
> I'm re-using the `Partner` enum that [`CloudInstance` uses](https://github.com/dockstore/dockstore/blob/0ff1f507610a0ddf125b2f3781c0bda3992a846e/dockstore-webservice/src/main/java/io/dockstore/webservice/core/CloudInstance.java#L47) because they seem to represent the same thing. We've always used `platform` when referring to where the workflow is executed when discussing metrics. I thought about changing the existing `Partner` enum name to `Platform`, but I think that will have ripple effects in the UI so I didn't. Do you think I should change the name in the description/field to `partner` instead?

No, I don't think it's necessary at this point. Maybe my mind will change in the future, but it's easy enough if we want to.

Reply:
I like the enum but agree we can expand it a lot (I like this being a controlled vocabulary but am open to adding some messy "other"/"grab-bag" categories for DIY efforts)

Reply:
I added AGC and OTHER to the enum